### DISCUSSION TITLE
Does APK appears in Play even though publish task fails?

### DISCUSSION THREAD

**Discussion by Steyrix** [Triple-T/gradle-play-publisher#938] [Q&A]
My problem is a bit complicated. I publish an app to internal track using this plugin via my Jenkins pipeline, however publishing fails 

    {
     "code" : 400,
     "errors" : [ {
     "domain" : "global",
     "message" : "This Edit has been deleted.",
     "reason" : "failedPrecondition"
      } ],
     "message" : "This Edit has been deleted.",
     "status" : "FAILED_PRECONDITION"
    }

I believe it happens due to publishing multiple APK variants at the same time. Anyways, **does APK still appears in the track even though pipeline fails**? I am unable to check it manually, since it is the first publishing to internal track, which is known to be very long to be synchronised by Google Play and appear in the track.

**Comment by SUPERCILEX** [comment]
No, if the edit is deleted then whatever changes were made can't be committed. The APK might have been uploaded, but it wouldn't be in a track. As a sidenote, you cannot run multiple builds that touch the same app package concurrently.

**Reply by Steyrix** [reply]
@SUPERCILEX , thank you for your answer! 
However, I need to know: what if I need to upload multiple APK's for different platforms? Is there any workaround for publishing?

**Reply by SUPERCILEX** [reply]
Publishing will work fine as long as you do it all in one go in the same build.
